### PR TITLE
Add swig dependency

### PR DESCRIPTION
**Pull request by gguo11837463**
We don't want to download swig manually.
Hope pip install can finish it directly.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -44,13 +44,13 @@ def run(self):
     name='pylibfdt',
     cmdclass = {'build_py' : build_py},
     use_scm_version=True,
-    setup_requires = ['setuptools_scm'],
+    setup_requires = ['setuptools', 'swig'],
```
Comment (sjg20) [inline setup.py]: Do you need the change to setuptools ?

**Review by sjg20 — CHANGES_REQUESTED** [review]
This needs to land in dtc upstream:

https://github.com/dgibson/dtc

But setup.py is a bit out-of-sync in the two projects, so I will see if I can tidy that up.

**Comment by robherring** [comment]
> This needs to land in dtc upstream:
> 
> https://github.com/dgibson/dtc
> 
> But setup.py is a bit out-of-sync in the two projects, so I will see if I can tidy that up.

Good luck with that. I gave up. The issue is that python has its own ideas about how to build things and that's at odds with make and meson (more meson). IMO, we shouldn't try to wrap the python build with make and meson, but let users interact with python build directly that anyone familiar with python packages understands (i.e. running ./setup.py or pip). And it's not just running the build/install, but all the other python tools which integrate in such as pytest. Or at a minimum, finish the meson conversion first, so there's not 2 environments you are trying to shoehorn setup.py into. Also, I'm sure you are aware, but setup.py has been superseded by setup.cfg^Wpyproject.toml. Might want to consider moving to that before wrestling with make/meson integration. I'm sure pyproject.toml has its own quirks to deal with.

The other issue was also what is the source of libfdt. Is it the one you build (and install?) or the distro installed one. IIRC, that was at odds with what I wanted to do here.

File: setup.py (hunk 1 above)
Comment (robherring) [reply setup.py]: AIUI, 'swig' here is just a wrapper package to download and install swig which is not a python package. What happens if the distro version of swig is already installed?

Comment (gguo11837463) [reply setup.py]: I use python 3.12 to verify it on Windows, setuptools is necessary.

Without this PR and don't install swig manually
- py -3.12 -m pip install git+https://github.com/devicetree-org/pylibfdt.git@main
image


With this PR and don't install swig manually.
- py -3.12 -m pip install git+https://github.com/gguo11837463/pylibfdt.git@gua_pylibfdt_depend_swig
image

**Comment by gguo11837463** [comment]
> > This needs to land in dtc upstream:
> > https://github.com/dgibson/dtc
> > But setup.py is a bit out-of-sync in the two projects, so I will see if I can tidy that up.
> 
> Good luck with that. I gave up. The issue is that python has its own ideas about how to build things and that's at odds with make and meson (more meson). IMO, we shouldn't try to wrap the python build with make and meson, but let users interact with python build directly that anyone familiar with python packages understands (i.e. running ./setup.py or pip). And it's not just running the build/install, but all the other python tools which integrate in such as pytest. Or at a minimum, finish the meson conversion first, so there's not 2 environments you are trying to shoehorn setup.py into. Also, I'm sure you are aware, but setup.py has been superseded by setup.cfg^Wpyproject.toml. Might want to consider moving to that before wrestling with make/meson integration. I'm sure pyproject.toml has its own quirks to deal with.
> 
> The other issue was also what is the source of libfdt. Is it the one you build (and install?) or the distro installed one. IIRC, that was at odds with what I wanted to do here.

I just use it https://pypi.org/project/pylibfdt/ by pip install pylibfdt, but if I don't install by chocolatey or https://sourceforge.net/p/swig/news/2024/02/swig-421-released/ on windows, pip install always failure, so want to make it done by this way.

**Comment by gguo11837463** [comment]
> > This needs to land in dtc upstream:
> > https://github.com/dgibson/dtc
> > But setup.py is a bit out-of-sync in the two projects, so I will see if I can tidy that up.
> 
> Good luck with that. I gave up. The issue is that python has its own ideas about how to build things and that's at odds with make and meson (more meson). IMO, we shouldn't try to wrap the python build with make and meson, but let users interact with python build directly that anyone familiar with python packages understands (i.e. running ./setup.py or pip). And it's not just running the build/install, but all the other python tools which integrate in such as pytest. Or at a minimum, finish the meson conversion first, so there's not 2 environments you are trying to shoehorn setup.py into. Also, I'm sure you are aware, but setup.py has been superseded by setup.cfg^Wpyproject.toml. Might want to consider moving to that before wrestling with make/meson integration. I'm sure pyproject.toml has its own quirks to deal with.
> 
> The other issue was also what is the source of libfdt. Is it the one you build (and install?) or the distro installed one. IIRC, that was at odds with what I wanted to do here.

I just use it https://pypi.org/project/pylibfdt/ by pip install pylibfdt, but if I don't install by chocolatey install swig or https://sourceforge.net/p/swig/news/2024/02/swig-421-released/ on windows, pip install always failure, so want to make it done by this way.

**Comment by sjg20** [comment]
I sent some patches upstream too

https://www.spinics.net/lists/devicetree-compiler/

**Review by sjg20 — APPROVED** [review]

**Pull request merged**